aptos-move/e2e-move-tests/src/tests/object_args.rs

```diff
@@ -0,0 +1,119 @@
+    0x90, 0x71, 0xAA, 0x3F, 0xBD, 0x2A, 0xB9, 0x51, 0x37, 0xF7, 0xCB, 0xAD, 0x13, 0x6F, 0x09, 0x2B,
+]);
+
+fn success(tests: Vec<(&str, Vec<Vec<u8>>, &str)>) {
+    success_generic(vec![], tests)
+}
+
+fn success_generic(ty_args: Vec<TypeTag>, tests: Vec<(&str, Vec<Vec<u8>>, &str)>) {
+    let mut h = MoveHarness::new_with_features(vec![FeatureFlag::STRUCT_CONSTRUCTORS], vec![]);
+
+    // Load the code
+    let acc = h.new_account_at(AccountAddress::from_hex_literal("0xcafe").unwrap());
+    assert_success!(h.publish_package(&acc, &common::test_dir_path("object_args.data/pack")));
+
+    let module_data = parse_struct_tag("0xCAFE::test::ModuleData").unwrap();
+
+    // Check in initial state, resource does not exist.
+    assert!(!h.exists_resource(acc.address(), module_data.clone()));
+
+    for (entry, args, expected_change) in tests {
+        assert_success!(h.run_entry_function(
+            &acc,
+            str::parse(entry).unwrap(),
+            ty_args.clone(),
+            args,
+        ));
+        assert_eq!(
+            String::from_utf8(
+                h.read_resource::<ModuleData>(&OBJECT_ADDRESS, module_data.clone())
+                    .unwrap()
+                    .state
+            )
+            .unwrap(),
+            expected_change,
+        );
+    }
+}
+
+fn fail(tests: Vec<(&str, Vec<Vec<u8>>, StatusCode)>) {
+    fail_generic(vec![], tests)
+}
+
+fn fail_generic(ty_args: Vec<TypeTag>, tests: Vec<(&str, Vec<Vec<u8>>, StatusCode)>) {
+    let mut h = MoveHarness::new_with_features(vec![FeatureFlag::STRUCT_CONSTRUCTORS], vec![]);
+
+    // Load the code
+    let acc = h.new_account_at(AccountAddress::from_hex_literal("0xcafe").unwrap());
+    assert_success!(h.publish_package(&acc, &common::test_dir_path("object_args.data/pack")));
+
+    let module_data = parse_struct_tag("0xCAFE::test::ModuleData").unwrap();
+
+    // Check in initial state, resource does not exist.
+    assert!(!h.exists_resource(acc.address(), module_data));
+
+    for (entry, args, err) in tests {
+        // Now send hi transaction, after that resource should exist and carry value
+        let status = h.run_entry_function(&acc, str::parse(entry).unwrap(), ty_args.clone(), args);
+        assert_vm_status!(status, err);
+    }
+}
+
+#[test]
+fn object_args_good() {
+    let tests = vec![
+        // ensure object exist
+        ("0xcafe::test::initialize", vec![], ""),
+        (
+            "0xcafe::test::object_arg",
+            vec![
+                bcs::to_bytes("hi").unwrap(),
+                bcs::to_bytes(&OBJECT_ADDRESS).unwrap(),
+            ],
+            "hi",
+        ),
+        (
+            "0xcafe::test::pass_optional_fixedpoint",
+            vec![
+                bcs::to_bytes(&OBJECT_ADDRESS).unwrap(),     // Object<T>
+                bcs::to_bytes(&vec![(1u64 << 32)]).unwrap(), // Option<FixedPoint>
+            ],
+            "4294967296",
+        ),
+    ];
+
+    success(tests);
+}
+
+#[test]
```

Comment:
Can we add a test for when the new constructors feature flag is disabled?

Reply:
What is there to test? We still test strings which continue to work.

Reply:
For when you try to send object<T> and the flag is disabled. This covers the same failure path discussed in the other comment

Reply:
If the flag is disabled then, the situation is as it is now. The other code path is not reached